### DISCUSSION TITLE
C# file showing up as plain text

### DISCUSSION THREAD

**Discussion by carlossanlop** [github-linguist/linguist#5318] [Classification]
A few files in this PR https://github.com/dotnet/roslyn-analyzers/pull/4806 are not getting C# syntax highlighting. Seems like a GitHub bug.

Creating the discussion first because that was the recommendation in the issue template.

**Comment by Alhadis** [comment]
Can you clarify *which* files in your pull-request are affected?

**Comment by lildude** [comment]
This sounds a lot like the problem previously reported in https://github.com/github/linguist/issues/5252 but in this case I can see it's slightly different. The files that don't show syntax highlighting are those that are below the fold and "lazy loaded" as you scroll.

From that PR, I start to see the problem from the fifth file:

![CleanShot 2021-04-09 at 09 23 52](https://user-images.githubusercontent.com/627280/114152108-c85f8c00-9915-11eb-9aa3-3614d327519b.png)

... and then _every_ file beyond that fails to show syntax highlighting, regardless of language. Syntax highlighting does however show when viewing the file in isolation. Using the split-view trick doesn't change the behaviour unlike in https://github.com/github/linguist/issues/5252

So in short...

> Seems like a GitHub bug.

Spot On!!! This appears to be a problem with the way the GitHub site is loading those files on large PRs. This is not anything Linguist has any control over so this is not the right place for this. I recommend you use the "Contact GitHub" link from the PR page showing the problem to contact GitHub support.

**Comment by carlossanlop** [comment]
> Can you clarify which files in your pull-request are affected?

@Alhadis the files that are showing in plain text, instead of with C#. It's pretty easy to spot. @lildude pointed them out already.

> But I recommend you use the "Contact GitHub" link from the PR page showing the problem to contact GitHub support.

@lildude, I initially opened the issue there, and they sent me here!

Also, Discussions are broken. I got zero notifications from this thread. I had to manually come back to see if there were any updates.

The whole process of logging GitHub is really frustrating.

**Reply by lildude** [reply]
@carlossanlop I've found your ticket and I can see why you were directed here: your initial problem description is essentially the first sentence in this discussion which is very vague and doesn't provide much to go on, hence the response you got and @Alhadis's question.

Unfortunately I can't comment on the ticket to provide further details myself so you'll need to do so. Please feel free to send my response above (with the link) and mention I can be contacted directly if they need any further clarification.

**Reply by lildude** [reply]
I can see this has now been resolved.
